websocket/_app.py

```diff
@@ -331,7 +344,7 @@ def teardown(close_frame=None):
             # Finally call the callback AFTER all teardown is complete
             self._callback(self.on_close, close_status_code, close_reason)
 
-        def setSock():
+        def setSock(reconnecting=False):
```

Comment:
What is this `reconnecting` variable useful for? I see it in line 370 but I don't think I'm clear on how its use on that line impacts the reconnect process. This variable name is very close to `reconnect`, so a different name that better differentiates would be better.

Reply:
That's so that on line 370 it doesn't call handleDisconnect() again if it's already "reconnecting". That's actually the heart of this pull request - in synchronous (default) mode, the stack grows if handleDisconnect() is called (by way of a couple intermediate functions) recursively. Thus, when DispatcherBase.reconnect() (line 51) calls reconnector() (which is setSock()), it passes reconnecting=True, so that if the connection fails it exits setSock(), returns to DispatcherBase.reconnect() (popping the setSock() frame off the stack), and tries again after the wait interval.

Reply:
And of course I'm open to changing "reconnecting" if there's a better name for that kwarg :)